src/utils/utils.ts

```diff
@@ -606,3 +606,16 @@ export function stringClamp(value: string, def: number, min: number, max: number
     const nanToNull = (nr: number): null | number => (isNaN(nr) ? null : nr)
     return clamp(nanToNull(parseInt(value)) ?? def, min, max)
 }
+
+export function extractElements(elements: Record<string, any>[]): Element[] {
```

Comment:
A bunch of element code is in `src/utils/db/utils.ts`. Perhaps this makes sense to put there as well?